tests/mlmodel_langchain/test_agent.py

```diff
@@ -0,0 +1,147 @@
+)
+
+from newrelic.api.background_task import background_task
+from newrelic.common.object_names import callable_name
+from newrelic.common.package_version_utils import get_package_version
+
+LANGCHAIN_VERSION = get_package_version("langchain")
+
+
+@pytest.fixture
+def tools():
+    @tool
+    def multi_arg_tool(first_num, second_num):
+        """A test tool that adds two integers together"""
+        return first_num + second_num
+
+    return [multi_arg_tool]
+
+
+@pytest.fixture
+def prompt():
+    return ChatPromptTemplate.from_messages(
+        [
+            (
+                "system",
+                "You are a world class algorithm for extracting information in structured formats.",
+            ),
+            (
+                "human",
+                "Use the given format to extract information from the following input: {input}",
+            ),
+            ("human", "Tip: Make sure to answer in the correct format"),
+            MessagesPlaceholder(
+                variable_name="agent_scratchpad",
+            ),
+        ]
+    )
+
+
+@pytest.mark.skip(reason="Streaming not enabled yet for NR")
+@reset_core_stats_engine()
+@validate_transaction_metrics(
+    name="test_agent:test_sync_agent",
+    scoped_metrics=[("Llm/agent/Langchain/invoke", 1)],
+    rollup_metrics=[("Llm/agent/Langchain/invoke", 1)],
+    custom_metrics=[
+        ("Python/ML/Langchain/%s" % LANGCHAIN_VERSION, 1),
+    ],
+    background_task=True,
+)
+@background_task()
+def test_sync_agent(chat_openai_client, tools, prompt):
+    agent = create_openai_functions_agent(chat_openai_client, tools, prompt)
+    agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)
+    response = agent_executor.invoke({"input": "Hello, world"})
+    assert response
+
+
+@pytest.mark.skip(reason="Streaming not enabled yet for NR")
+@reset_core_stats_engine()
+@validate_transaction_metrics(
+    name="test_agent:test_async_agent",
+    scoped_metrics=[("Llm/agent/Langchain/ainvoke", 1)],
+    rollup_metrics=[("Llm/agent/Langchain/ainvoke", 1)],
+    custom_metrics=[
+        ("Python/ML/Langchain/%s" % LANGCHAIN_VERSION, 1),
+    ],
+    background_task=True,
+)
+@background_task()
+def test_async_agent(loop, chat_openai_client, tools, prompt):
+    agent = create_openai_functions_agent(chat_openai_client, tools, prompt)
+    agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)
+    loop.run_until_complete(agent_executor.ainvoke({"input": "Hello, world"}))
+
+
+@pytest.mark.skip(reason="Streaming not enabled yet for NR")
+@reset_core_stats_engine()
+@validate_transaction_error_event_count(1)
+@validate_error_trace_attributes(
+    callable_name(KeyError),
+)
+@validate_transaction_metrics(
+    name="test_agent:test_sync_agent_error",
+    custom_metrics=[
+        ("Python/ML/Langchain/%s" % LANGCHAIN_VERSION, 1),
+    ],
+    background_task=True,
+)
+@background_task()
+def test_sync_agent_error(chat_openai_client, tools, prompt):
```

Comment:
I don't think either of these error tests are necessary. They are already covered by chain_tests. Let's remove these two tests.